gs-rest-client/src/main/java/client/SimpleClient.java

```diff
@@ -118,95 +118,132 @@ public static void fixMyBug(ServerRequest serverRequest, TokenizerBuilder bobThe
                 e.printStackTrace();
             }
 
-            //Read JSON response from the server in a BufferedReader
+            //Read the byte response from the server in a BufferedReader
             BufferedReader br = new BufferedReader(new InputStreamReader((conn.getInputStream())));
-
             String output;
-            System.out.println("Output from Server .... \n\n");
+
+            //Convert input stream of bytes into a JSON string
             StringBuilder returnedJsonStringBuilder = new StringBuilder();
             while ((output = br.readLine()) != null) {
                 returnedJsonStringBuilder.append(output).append("\n");
             }
             returnedJsonStringBuilder.setLength(returnedJsonStringBuilder.length() - 1); //remove extra newline
             String returnedJsonString = returnedJsonStringBuilder.toString();
 
-            if (method.equals("index")) {
-                System.out.println(returnedJsonString.toString());
-                conn.disconnect();
+            //Handle the server's response based on the requested method.
+            handleResponse(returnedJsonString, tokenBuilder, method);
```

Comment:
Added a new method with the sole job of handling the JSON string response from the server. This removes any handling of responses in the REQUEST method.